From dbbac5a04c7cec1c14026e807e998eae387ddee4 Mon Sep 17 00:00:00 2001
From: Andrei Borzenkov <arvidjaar@gmail.com>
Date: Sun, 30 Nov 2014 18:48:42 +0300
Subject: [PATCH 125/506] grub-core/lib/syslinux_parse.c: do not free array

say->msg is inline array in a structure and should not be freed.
CID: 73610
---
 ChangeLog                      | 5 +++++
 grub-core/lib/syslinux_parse.c | 1 -
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index b6220de..21434ff 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2014-11-30  Andrei Borzenkov  <arvidjaar@gmail.com>
+
+	* grub-core/lib/syslinux_parse.c (free_menu): Do not free
+	inline array (CID 73610).
+
 2014-11-28  Andrei Borzenkov  <arvidjaar@gmail.com>
 
 	* grub-core/io/lzopio.c (test_header): Fix double free (CID 73665)
diff --git a/grub-core/lib/syslinux_parse.c b/grub-core/lib/syslinux_parse.c
index 6bc504b..08e451c 100644
--- a/grub-core/lib/syslinux_parse.c
+++ b/grub-core/lib/syslinux_parse.c
@@ -1351,7 +1351,6 @@ free_menu (struct syslinux_menu *menu)
   for (say = menu->say; say ; say = nsay)
     {
       nsay = say->next;
-      grub_free (say->msg);
       grub_free (say);
     }
 
-- 
2.4.3

